src/main/java/org/apache/commons/crypto/utils/ReflectionUtils.java

```diff
@@ -63,6 +68,9 @@ private static abstract class AbstractNegativeCacheSentinel {
     public static Class<?> getClassByName(final String name) throws ClassNotFoundException {
         final Class<?> ret = getClassByNameOrNull(name);
         if (ret == null) {
+            if (INIT_ERROR_CLASSES.get(CLASSLOADER).contains(name)) {
+                throw new IllegalStateException("Class " + name + " initialization error");
```

Comment:
An additional judgment has been added to distinguish ClassNotFoundException. Here an `IllegalStateException` is manually thrown, there may be a more appropriate exception type.